feat: file upload feature

35C4n0r opened the pull request.

BrunoQuaresma commented:
From a UX perspective, I think showing a dialog after clicking the upload button feels different from how similar tools usually behave. IMO, clicking the upload button should immediately open the file picker. Drag and drop could happen directly on the screen or be tied to the “prompt area.” This would align more closely with the approach used by ChatGPT or Gemini.

That said, I’m not sure what you’ve already aligned on with the PMs. cc: @matifali

35C4n0r commented on lib/httpapi/server.go:
```diff
@@ -391,6 +400,76 @@ func (s *Server) createMessage(ctx context.Context, input *MessageRequest) (*Mes
+	}
+
+	for _, f := range zipReader.File {
+		outPath := filepath.Join(formData.UploadPath, f.Name)
```
addressed, the path looks something like this now 
@"/tmp/agentapi-uploads-2444351403/8e389968c5207d22/awsad.png"
where "8e389968c5207d22" is part of the checksum of the file.

35C4n0r commented:
> From a UX perspective, I think showing a dialog after clicking the upload button feels different from how similar tools usually behave. IMO, clicking the upload button should immediately open the file picker. Drag and drop could happen directly on the screen or be tied to the “prompt area.” This would align more closely with the approach used by ChatGPT or Gemini.
> 
> That said, I’m not sure what you’ve already aligned on with the PMs. cc: @matifali

@BrunoQuaresma addressed the comment, requesting a re-review

35C4n0r commented:
@BrunoQuaresma thanks for sharing the [blog](https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript), it was very helpful.
I have removed the `any`s from the error and handled the ts properly.

johnstcn commented on lib/httpapi/server.go:
```diff
@@ -417,6 +436,45 @@ func (s *Server) createMessage(ctx context.Context, input *MessageRequest) (*Mes
+
+	file := formData.File.File
+
+	buf, err := io.ReadAll(file)
```
Can we use an `io.LimitReader` here? Let's set a limit of say 10MB on a file upload.

35C4n0r replied:
addressed 👍🏻

johnstcn commented on lib/httpapi/server_test.go:
```diff
@@ -652,3 +655,188 @@ func assertSSEHeaders(t testing.TB, resp *http.Response) {
+			fileContent:        "content",
+			expectedStatusCode: http.StatusOK,
+			expectFilePath:     true,
+		},
```
Can you add test cases with absolute or relative filenames?

e.g. `/bin/bash` or `../../../../initrd.img`

35C4n0r replied:
addressed 👍🏻
